elasticdl/pkg/kernel/kernel_test.go

```diff
@@ -25,3 +26,58 @@ func TestSGD(t *testing.T) {
 
 	assert.Equal(t, b, expected)
 }
+
+func TestAdam(t *testing.T) {
+	const size int = 10
+	grad := make([]float32, size)
+	param := make([]float32, size)
+	m := make([]float32, size)
+	v := make([]float32, size)
+	var lr float64 = 0.1
+	var step int64 = 5
+	var beta1 float64 = 0.9
+	var beta2 float64 = 0.999
+	var epsilon float64 = 1e-8
+
+	for i := 0; i < size; i++ {
+		grad[i] = rand.Float32()
+		param[i] = rand.Float32()
+		m[i] = rand.Float32()
+		v[i] = rand.Float32()
+	}
+
+	expectedParam := make([]float32, size)
+	expectedM := make([]float32, size)
+	expectedV := make([]float32, size)
+
+	for i := 0; i < size; i++ {
+		expectedM[i] = float32(beta1)*m[i] + (1-float32(beta1))*grad[i]
+		expectedV[i] = float32(beta2)*v[i] + (1-float32(beta2))*grad[i]*grad[i]
+	}
+
+	for i := 0; i < size; i++ {
+		expectedParam[i] = param[i] - float32(lr)*expectedM[i]/(1-float32(math.Pow(beta1, float64(step))))/(float32(math.Sqrt(float64(expectedV[i]/(1-float32(math.Pow(beta2, float64(step))))))+epsilon))
```

Comment:
This Line is too long.  Try `gofmt -w kernel_test.go`?

Reply:
I used the pre-commit run, do I still need to manually run this command?

Reply:
I am not sure why `pre-commit` is not worked as expected. I will debug on this.
You could run `gofmt` manually in this PR.

Reply:
Go does not limit line length. It is up to the user to make it reasonable.